components/Form/Form.js

```diff
@@ -20,71 +21,90 @@ export default function FormToCreateShoppingItem({
+            type="text"
+            required
+            placeholder="e.g., salt"
+            defaultValue={item.name || ""}
+          />
+          <StyledLabel htmlFor="quantity">quantity*:</StyledLabel>
+          <StyledInput
+            id="quantity"
+            name="quantity"
+            type="number"
+            required
+            defaultValue={item.quantity || ""}
+          />
+          <StyledLabel htmlFor="category">category*:</StyledLabel>
+          <select
+            id="category"
+            name="category"
+            // data-js="category"
+            required
+            defaultValue={
+              // I have replaced the line below with "item.id" because it didn't work anymore after the suggested refactoring;
+              // probably because I had undone handing down the categories(?)
+              // mode === "edit"
+              // item.id
+              //   ?
+              item.category || ""
+              // :
+              // replace "" by "default" or undo "" > "default" elsewhere
+              // ""
+            }
+          >
+            <option value="">please select a category</option>
+            {categories.map((category) => (
+              <option key={category.name} value={category.name}>
+                {category.name}
+              </option>
+            ))}
+            ;
+          </select>
 
-          <StyledLabel>
-            category*:
-            <select
-              key="category"
-              name="category"
-              data-js="category"
-              required
-              defaultValue={mode === "edit" ? item.category : ""}
-            >
-              <option value="">please select a category</option>
-              {categories.map((category) => (
-                <option key={category.name} value={category.name}>
-                  {category.name}
-                </option>
-              ))}
-              ;
-            </select>
-          </StyledLabel>
-          <StyledLabel>
-            comment:
-            <StyledInput
-              name="comment"
-              type="text"
-              defaultValue={item.comment || ""}
-              placeholder="Enter comments here..."
-            />
-          </StyledLabel>
+          <StyledLabel htmlFor="comment">comment:</StyledLabel>
+          <StyledInput
+            name="comment"
+            type="text"
+            defaultValue={item.comment || ""}
+            placeholder="preferably sea salt"
+          />
           <StyledNote>
             Required fields are followed by <span aria-label="required">*</span>
             .
           </StyledNote>
-          <StyledButton>Submit</StyledButton>
+          <StyledButton>
+            submit
+            {/* {submitLabel} */}
+          </StyledButton>
           <button
             type="button"
-            onClick={() => onChangeMode(mode === ("add" || "edit") && "")}
+            onClick={() => onChangeMode(mode === "default")}
```

Comment:
The `onChangeMode` function receives as a parameter the new value, which should be saved in your mode state, e.g. "default".
```suggestion
            onClick={() => onChangeMode("default")}
```

Reply:
Done